app/Http/Controllers/Admin/VoucherController.php

```
@@ -123,25 +137,273 @@ public function purchases(Request $request)
+            'shared_users' => 'nullable|integer|min:1',
+            'session_timeout' => 'nullable|string|max:20',
+            'price' => 'nullable|numeric|min:0',
+            'agent_price' => 'nullable|numeric|min:0',
+            'validity' => 'nullable|string|max:20',
+            'is_active' => 'boolean',
+        ]);
+
+        $mikrotik = app(MikrotikService::class);
+        $speeds = $mikrotik->parseRateLimit($validated['rate_limit'] ?? '');
+
+        $profile = HotspotProfile::create([
```

PHPMD (2.15.0)

Avoid unused local variables such as '$profile'. (undefined)

(UnusedLocalVariable)